lib/Parse/ParseDecl.cpp

```diff
@@ -5435,10 +5435,29 @@ static void skipAttribute(Parser &P) {
 
 bool Parser::isStartOfSwiftDecl(bool allowPoundIfAttributes,
                                 bool hadAttrsOrModifiers) {
+  const bool isTopLevelLibrary = (SF.Kind == SourceFileKind::Library) ||
+                                 (SF.Kind == SourceFileKind::Interface) ||
+                                 (SF.Kind == SourceFileKind::SIL);
   if (Tok.is(tok::at_sign) && peekToken().is(tok::kw_rethrows)) {
     // @rethrows does not follow the general rule of @<identifier> so
     // it is needed to short circuit this else there will be an infinite
     // loop on invalid attributes of just rethrows
+  } else if ((Tok.getKind() == tok::identifier) &&
```

Comment:
Should this logic be gated behind your experimental feature flag?

Reply:
ah yes indeed...